spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -164,7 +164,44 @@ Example query methods:
 - `findByAuthorName(String authorName)`: Find entities by the 'authorName' property of a related entity.
 - `findByCategoryNameAndPriceLessThan(String categoryName, double price)`: Find entities by 'categoryName' and 'price' properties, applying an 'And' condition.
 
-
+==== BNF Grammar for Query Methods
+
+Query methods allow developers to create database queries using method naming conventions. These methods consist of a subject, predicate, and optional order clause. This BNF notation provides a structured representation for understanding and implementing these powerful querying techniques in your applications.
+
+[source,bnf]
+----
+<query-method> ::= <subject> <predicate> [<order-clause>]
+<subject> ::= <action> "By" | <action> <expression> "By"
+<action> ::= "find" | "delete" | "update" | "count" | "exists"
+<expression> ::= <identifier> | <identifier> <number>
+<predicate> ::= <condition> { ("And" | "Or") <condition> }
+<condition> ::= <property> <operator>
+<operator> ::= "Equals" | "Is" | "LessThan"| "LessThanEqual" | "GreaterThan" | "GreaterThanEqual" | "Between" | "Empty" | "Like" | "IgnoreCase" | "In" | "Null" | "True" | "False"
+<property> ::= <identifier> | <identifier> "_" <property>
+<identifier> ::= <word>
+<number> ::= <integer> | <float>
+<integer> ::= <digit> { <digit> }
+<float> ::= <integer> "." <integer>
+<order-clause> ::= "OrderBy" <order-item> { "," <order-item> }
+<order-item> ::= <property> ["Asc" | "Desc"]
+----
+
+Explanation of the BNF elements:
+
+- `<query-method>`: Represents a query method, which consists of a subject, a predicate, and an optional order clause.
+- `<subject>`: Defines the action (e.g., "find" or "delete") followed by an optional expression and "By."
+- `<action>`: Specifies the action, such as "find" or "delete."
+- `<expression>`: Represents an optional expression, such as "First10."
```

Comment:
```suggestion
- `<find-expression>`: Represents an optional expression for find operations, such as "First10."
```
The specification only defines usage of the "First" expression for find...By, so the BNF needs to limit it to find...By.  If we want to consider requiring for others, like countFirst10By or deleteFirst10By, that would be a separate pull.